config/checkstyle_non_main_files_checks.xml

```diff
@@ -24,6 +24,16 @@
     <message key="regexp.filename.mismatch"
       value="CI folder should only contain executables files"/>
   </module>
+  <module name="RegexpOnFilename">
+    <property name="id" value="regexpOnScriptsAndConfigs"/>
+    <property name="folderPattern" value="([\\/].ci([\\/]|$)|[\\/].github[\\/]workflows([\\/]|$))"/>
```

Comment:
Please put `name="fileExtensions" value="sh, groovy, cmd,`
As next line after folderPatten.

Should folder pattern be just ".ci" folder?

Reply:
> Should folder pattern be just ".ci" folder?

https://github.com/checkstyle/checkstyle/issues/12716#issue-1570511229
> Make GitHub workflow filenames consistent 
> The .github/workflows folder is inconsistent 

It doesn't look like the issue is about the CI folder. Why are we including the CI folder?

Reply:
> We should use kebab-case for naming of scripts or configs.

This is a bit broad now that I see. I suppose this means files in `config` folder as well, right?

Reply:
I think the issue is getting out of hand now and needs to be explicit what it is doing. I wouldn't expect this to cover the `config` folder.

@romani ping

Reply:
Yes, better to agree on what we fix, and it is completely fine to make several PRs to enforce rule on folders one by one or ... But not like big bang PR.

Reply:
I suggest we keep this one with `.github/workflows` folder only as the original issue was. Do you agreee?

Reply:
Yes